src/screen_event_handler.cpp

```diff
@@ -47,21 +50,47 @@ bool usc::ScreenEventHandler::handle(MirEvent const& event)
     if (input_event_type == mir_input_event_type_key)
     {
         auto const kev = mir_input_event_get_keyboard_event(input_event);
-        if (mir_keyboard_event_scan_code(kev) == KEY_POWER)
+        int key_code = mir_keyboard_event_scan_code(kev);
+        auto const action = mir_keyboard_event_action(kev);
+
+        printf("  kev code: %d\n", key_code);
+        if (key_code == KEY_POWER)
         {
-            auto const action = mir_keyboard_event_action(kev);
             if (action == mir_keyboard_action_down)
                 power_button_event_sink->notify_press();
             else if (action == mir_keyboard_action_up)
                 power_button_event_sink->notify_release();
         }
         // we might want to come up with a whole range of media player related keys
-        else if (mir_keyboard_event_scan_code(kev) == KEY_VOLUMEDOWN||
-                 mir_keyboard_event_scan_code(kev) == KEY_VOLUMEUP)
+        else if (
+                 // can come from headset button
+                    key_code == KEY_VOLUMEDOWN
+                 || key_code == KEY_VOLUMEUP
+                 || key_code == KEY_MEDIA
+
+                 // can come from gestures of touchpanel
+                 || key_code == KEY_NEXTSONG
+                 || key_code == KEY_PREVIOUSSONG
+                 || key_code == KEY_PLAYPAUSE
+                 || key_code == KEY_CAMERA
+                 || key_code == KEY_ATTENDANT_TOGGLE)
+
         {
-            // do not keep display on when interacting with media player
+            // do not keep display on for these gestures
+            // notify gestures on 'up'
+            if (action == mir_keyboard_action_up) {
+                switch(key_code) {
+                // volume up/down are already taken care of it seems
+                case KEY_MEDIA: gesture_event_sink->notify_gesture("media"); break;
+
+                case KEY_NEXTSONG: gesture_event_sink->notify_gesture("next-song"); break;
+                case KEY_PREVIOUSSONG: gesture_event_sink->notify_gesture("previous-song"); break;
+                case KEY_PLAYPAUSE: gesture_event_sink->notify_gesture("play-pause"); break;
+                case KEY_ATTENDANT_TOGGLE: gesture_event_sink->notify_gesture("toggle-flash"); break;
```

Comment:
You may want to put the `break` statements on the line after the `case` statements. This makes them easier to see when skimming the code.